### ISSUE TITLE
Filter operator for case-insensitive for string

### ISSUE THREAD

**Issue by dcube9** [alirezanet/Gridify#21]
It's usual to filter string data in case-insensitive mode.

A way to do that is add `#` operator.

`var gQuery = new GridifyQuery()
{
    Filter = "FirstName=#John, Address=*#st",
    OrderBy = "PhoneNumber"
};
`

Regards

**Comment by alirezanet** [comment]
Hi, thank you for your feedback
For now, Gridify doesn't have a case-insensitive operator yet but you can achieve case-insensitive search using the `value convertors`.

e.g
```c#
 var gq = new GridifyQuery { Filter = "name=BOB" };
 var gm = new GridifyMapper<TestClass>()
            .AddMap("name", q => q.Name.ToLower() , c => c.ToLower());

         var query = Ds.ApplyFiltering(gq, gm);
```

Although I agree having an operator is simpler, I'll look into this more.

**Comment by dcube9** [comment]
Hi, thank you for reply.

About symbol, I choice that without a specific reason.
The main problem (IMHO), is the string delimiter, every solution (# or /i or ...) have potentially a syntax problem.
But If we use a delimiter like ' (Apostrophe) for string in string you can write :
`var gQuery = new GridifyQuery() { Filter = "FirstName='John'/i, Address=*st/i", OrderBy = "PhoneNumber" };`
or
`var gQuery = new GridifyQuery() { Filter = "FirstName=#'John', Address=*#'st'", OrderBy = "PhoneNumber" };`

and when you need an apostrophe us \' .

what do you think?

**Comment by alirezanet** [comment]
> The main problem (IMHO), is the string delimiter, every solution (# or /i or ...) have potentially a syntax problem.

You are completely right, but I think supporting the Apostrophe characters has the same problem. also because Gridify is being used in a lot of projects I should avoid any major breaking changes.
I remember I had this idea to support (Apostrophe)  in version 1 but after few tests, I realized it's not helping too much.
this problem already exists for `( ) , |`  characters but it is easy enogth to solve.  `Regex.Replace(value, "([(),|])", "\\$1" )`. 
probably I'll only add an operator for case-insensitive support like `# or /i` ... 
thank you for the contribution

**Comment by alirezanet** [comment]
This feature is available in `v2.1.0` . 
```c#
var gq = new GridifyQuery() { Filter = "FirstName=John/i" };
```
